src/main/frontend/cli.py

```diff
@@ -413,80 +112,180 @@ def color_scores(scores: list[list[Dependency, float]]) -> \
     return colored_scores
 
 
-def analyze_sbom(args: Namespace) -> None:
+def parse_requirements(**kwargs) -> UserRequirements:
     """
-    Executes the command that analyzes an SBOM.
+    Parses the requirements from the arguments.
 
     Args:
-        args (Namespace): The parsed arguments of the program.
+        **kwargs (Any): The arguments.
+
+    Returns:
+        UserRequirements: The parsed user requirements.
     """
-    output, verbose, backend = parse_arguments_shared(args)
-    path, requirements = parse_analyze_arguments(args)
+    code_vulnerabilities: int = kwargs.get("code_vulnerabilities")
+    maintenance: int = kwargs.get("maintenance")
+    continuous_testing: int = kwargs.get("continuous_testing")
+    source_risk_assessment: int = kwargs.get("source_risk_assessment")
+    build_risk_assessment: int = kwargs.get("build_risk_assessment")
+
+    return UserRequirements({
+        RequirementsType.CODE_VULNERABILITIES: code_vulnerabilities,
+        RequirementsType.MAINTENANCE: maintenance,
+        RequirementsType.CONTINUOUS_TESTING: continuous_testing,
+        RequirementsType.SOURCE_RISK_ASSESSMENT: source_risk_assessment,
+        RequirementsType.BUILD_RISK_ASSESSMENT: build_risk_assessment,
+    })
+
+
+def validate_backend(_ctx, _param, value: str):
+    """
+    Validates that the backend URL is a valid URL.
+    """
+    value = value.replace("localhost", "127.0.0.1")
+    validation_res = validators.url(value)
+    if isinstance(validation_res, validators.ValidationError):
+        print(validation_res)
+        raise click.BadParameter("Invalid backend URL.")
+    else:
+        return value
+
+
+def validate_git_token(_ctx, _param, value: str):
+    """
+    Validates that a GitHub Personal Access Token is valid.
+    """
+    url = "https://api.github.com/user"
+    headers = {"Authorization": f"token {value}"}
+    response = requests.get(url, headers=headers, timeout=5)
+
+    # TODO: Could do more extensive error-checking here.
+    #       For example, different messages for different status codes.
+    if response.status_code != 200:
+        raise click.BadParameter("Failed to authenticate token. "
```

Comment:
Could include more granularity for status codes.
There is a function in the branch "fix-sbom-not-parsing" in util.py that could help with the statuscode 403 (out of token requests)

Reply:
This has been added